airflow/contrib/hooks/salesforce_hook.py

```diff
@@ -0,0 +1,471 @@
+
+        # define a map of strings to different schema converting functions
+        self.SF_TO_SCHEMA_MAP = {
+            "BQ": self.convertSalesforceSchemaToBQ
+        }
+
+    def signIn(self):
+        """
+        Sign into Salesforce.
+
+        If we have already signed it, this will just return the original object
+        """
+        if hasattr(self, 'sf'):
+            return self.sf
+
+        # connect to Salesforce
+        sf = Salesforce(
+            username=self.connection.login,
+            password=self.connection.password,
+            security_token=self.extras['security_token'],
+            instance_url=self.connection.host
+        )
+        self.sf = sf
+        return sf
+
+    def makeQuery(self, query):
+        """
+        Make a query to Salesforce.  Returns result in dictionary
+
+        :param query:    The query to make to Salesforce
+        """
+        if not hasattr(self, 'sf'):
+            self.signIn()
+
+        logging.info("Querying for all objects")
+        query = self.sf.query_all(query)
+
+        logging.info("Received results: Total size: \
+            {0}; Done: {1}".format(query['totalSize'], query['done']))
+
+        query = json.loads(json.dumps(query))
+        return query
+
+    def describeObject(self, obj):
+        """
+        Get the description of an object from Salesforce.
+
+        This description is the object's schema
+        and some extra metadata that Salesforce stores for each object
+
+        :param obj:     Name of the Salesforce object
+                        that we are getting a description of.
+        """
+        if not hasattr(self, 'sf'):
+            self.signIn()
+
+        return json.loads(json.dumps(self.sf.__getattr__(obj).describe()))
+
+    def getAvailableFields(self, obj):
+        """
+        Get a list of all available fields for an object.
+
+        This only returns the names of the fields.
+        """
+        if not hasattr(self, 'sf'):
+            self.signIn()
+
+        desc = self.describeObject(obj)
+
+        return [f['name'] for f in desc['fields']]
+
+    def _buildFieldList(self, fields):
+        # join all of the fields in a comma seperated list
+        return ",".join(fields)
+
+    def getObjectFromSalesforce(self, obj, fields):
+        """
+        Get all instances of the `object` from Salesforce.
+        For each model, only get the fields specified in fields.
+
+        All we really do underneath the hood is run:
+            SELECT <fields> FROM <obj>;
+        """
+        field_string = self._buildFieldList(fields)
+
+        query = "SELECT {0} FROM {1}".format(field_string, obj)
+        logging.info("Making query to salesforce: \
+            {0}".format(query if len(query) < 30 else " ... ".join(
+                [query[:15], query[-15:]])
+            )
+        )
+        return self.makeQuery(query)
+
+    def convertSalesforceSchemaToAnotherSchema(
```

Comment:
Ya, I prefer having this logic outside of the hook because it's really BQ related. It seems to me like you'd want this code to be part of a SalesforceToBigQuery operator. If someone wrote a SalesforceToRedshift operator, they'd write their own schema conversion logic there.